privacy-on-beam/pbeam/aggregations_test.go

```diff
@@ -251,3 +282,149 @@ func TestBoundedSumFloat64FnExtractOutputReturnsNilForSmallPartitions(t *testing
+		makePairsWithFixedVStartingFromKey(7+52, 99, 2),
+		makePairsWithFixedVStartingFromKey(7+52, 99, 2),
+		makePairsWithFixedVStartingFromKey(7+52, 99, 2),
+	)
+
+	_, s, col, want := ptest.CreateList2(pairs, result)
+	want = beam.ParDo(s, pairToKV, want)
+	col = beam.ParDo(s, pairToKV, col)
+	partitions := []int{0, 2}
+
+	partitionsCol := beam.CreateList(s, partitions)
+	epsilon, delta := 50.0, 1e-200
+	pcol := MakePrivate(s, col, NewPrivacySpec(epsilon, delta))
+	_, partitionT := beam.ValidateKVType(pcol.col)
+	got := dropUnspecifiedPartitionsVFn(s, partitionsCol, pcol, partitionT)
+	if err := equalsKVInt(s, got, want); err != nil {
+		t.Fatalf("dropUnspecifiedPartitionsVFn: for %v got: %v, want %v", col, got, want)
+	}
+}
+
+// Check that int elements with unspecified partitions
+// are dropped (tests function used for sum and mean).
+func TestDropUnspecifiedPartitionsInt(t *testing.T) {
+	triples := concatenateTriplesWithIntValue(
+		makeDummyTripleWithIntValue(7, 0),
+		makeDummyTripleWithIntValue(58, 1),
+		makeDummyTripleWithIntValue(99, 2),
+		makeDummyTripleWithIntValue(45, 100),
+		makeDummyTripleWithIntValue(20, 33))
+	// Keep partitions 0,2;
+	// Drop partitions 1, 33, 100;
+	result := concatenateTriplesWithIntValue(
+		makeDummyTripleWithIntValue(7, 0),
+		makeDummyTripleWithIntValue(99, 2))
+
+	_, s, col, col2 := ptest.CreateList2(triples, result)
+	// Doesn't matter that the values 3, 4, 5, 6, 9, 10
+	// are in the partitions PCollection because we are
+	// just dropping the values that are in our original PCollection
+	// that are not specified.
+	partitionsCol := beam.CreateList(s, []int{0, 2, 3, 4, 5, 6, 9, 10})
+	col = beam.ParDo(s, extractIDFromTripleWithIntValue, col)
+	col2 = beam.ParDo(s, extractIDFromTripleWithIntValue, col2)
+	epsilon, delta := 50.0, 1e-200
+
+	pcol := MakePrivate(s, col, NewPrivacySpec(epsilon, delta))
+	pcol = ParDo(s, tripleWithIntValueToKV, pcol)
+	got := dropUnspecifiedPartitionsKVFn(s, partitionsCol, pcol, pcol.codec.KType)
+	got = beam.SwapKV(s, got)
+
+	pcol2 := MakePrivate(s, col2, NewPrivacySpec(epsilon, delta))
+	pcol2 = ParDo(s, tripleWithIntValueToKV, pcol2)
+	want := pcol2.col
+	want = beam.SwapKV(s, want)
+
+	if err := equalsKVInt(s, got, want); err != nil {
+		t.Fatalf("DropUnspecifiedPartitionsInt: for %v got: %v, want %v", col, got, want)
+	}
+}
+
+// Check that float elements with unspecified partitions
+// are dropped (tests function used for sum and mean).
+func TestDropUnspecifiedPartitionsFloat(t *testing.T) {
+	// In this test, we check  that unspecified partitions
+	// are dropped. This function is used for sum and mean.
+	triples := concatenateTriplesWithFloatValue(
+		makeTripleWithFloatValue(7, 0, 2.0),
+		makeTripleWithFloatValueStartingFromKey(7, 100, 1, 1.3),
+		makeTripleWithFloatValueStartingFromKey(107, 150, 1, 2.5),
+		makeTripleWithFloatValueStartingFromKey(320, 150, 100, 5.5),
+	)
+	// Keep partition 0;
+	// drop partition 1.
```

Comment:
This is the same input/output structure used in the mean_test.go function ' TestMeanPerKeyNoNoiseFloatValues'... Since this is a different function I think it makes sense to use the test that I have... I can try to make that clearer in my comment.

Reply:
https://github.com/google/differential-privacy/blob/main/privacy-on-beam/pbeam/mean_test.go#L382